### ISSUE TITLE
Voyager should not break userspace again

### ISSUE THREAD

**Issue by damms005** [thedevdojo/voyager#3111]
- Laravel Version: 5.6
- Voyager Version: 1.1
- PHP Version:
- Database Driver & Version: Server version: 10.1.31-MariaDB Source distribution

### Description:
I will simply [quote the great Linus Tovalds](https://lkml.org/lkml/2012/12/23/75): 

> If a change results in user programs breaking, it's a bug

Simple. It is a very simple rule that keeps the Linux kernel dev community strong. You just simply don't roll and update that will break userspace.

Particularly for Voyager, devs have trusted that Voyager provides great allowance to manipulate the `User` model to app needs. Any change to this very important model is crucial because Laravel itself holds the `User` model in a very high esteem - auth facades and all. For Voyager to now create a crappy buggy `vendor/tcg/voyager/resources/views/users` folder that breaks this trust is not only a misnormal, it is userspace-breaking and heart-breaking. **You don't break userspace**. It is a very simple rule for any project that depends on developers's usage to become a great project.

We understand that you intend this change to be a new, cool feature of Voyager. But it breaks userspace. It is a very bad one.

The only workaround for now is to delete the crappy `vendor/tcg/voyager/resources/views/users` folder so that the sweet, old, powerful feature of a malleable, flexible, and powerful `User` model is restored

**Comment by damms005** [comment]
Please let no further update/changes to this great Voyager package depend on `vendor/tcg/voyager/resources/views/users` being available (since the current workaround breaks that thumb rule of composer dependency: _do not edit vendor package folder_)

Thank you

**Comment by fletch3555** [comment]
`Userspace` is a concept that is fairly restricted to the *nix kernel, and really doesn't apply to web applications.  There are similar concepts that do apply here, however, called [semver](https://semver.org/).  The great news is that it's 100% up to the project maintainer(s) to decide whether to follow semver or not.  For us, we (loosely) follow a modified version of it.  Instead of restricting breaking changes to the `major` releases (1.0.0, 2.0.0, etc), we have chosen to restrict them to `minor` releases (1.1.0, 1.2.0, etc).  This is similar to what Laravel does, which is the platform this package is built on top of.

To speak to your issue in particular.  The `user` view was updated in 1.1.0, which follows our versioning schema.  It does NOT break your ability to dynamically change the form, and you are NOT required to "delete the crappy [vendor] folder".  The users `edit-add` view is overridden no different than how our documentation tells you to do so for your own project.  This was done to prevent autocompletion of the password field (#1993, 2909).

If you REALLY want it to function like any other new BREAD type, then just [override the view per our documentation](https://voyager.readme.io/docs/overriding-views).  You can copy `vendor/tcg/voyager/resources/views/bread/edit-add.blade.php` to `resources/views/vendor/voyager/users/edit-add.blade.php`.

Your issue is unnecessarily confrontational and not at all professional.  Something like this should be mentioned on our slack group so we can have a conversation about it.  Please consider this a warning regarding your tone and demeanor.

Tl;Dr - We'll manage the project however we choose to.  Watch your tone.

**Comment by damms005** [comment]
You apparently misunderstood my intention. I do not intend to be rude or demean in any way. Voyager is a very good project and I am simply positing out of the zeal I have for it to keep moving forward

This should not be misunderstood for hate please.

I will like to see Voyager continuously take the lead in the Laravel Admin niche - that is why I was outspoken and blunt in my critic, albeit I apparently I'm oblivious of the fact that the issue can be reverted to previous use-case without necessary coding model editing field and all.

Thanks also for the cue on using the slack community more

**Comment by fletch3555** [comment]
No problem.  I appreciate the critique and nudge toward "best practice".  I doubt the linux kernel is the best comparison, but I appreciate it nonetheless.

Definitely come join us on the Slack group.  It's not SUPER active, but we're around.  Even if I'm not commenting, I definitely read every message.

**Comment by damms005** [comment]
Yeah. Definitely will join the Slack group. About the Linux kernel, well, I appreciate the heat in that space, as well as the spirit of Torvalds. Those things can help drive things, albeit roughly.

I am working on a PR to intelligently handle some things at the Voyager front-end too. Will be happy to share with the Slack group (e.g. like the good work done in v1.1 update wherein Textboxes are  pre-selected for columns when adding new BREAD to model).
